### DISCUSSION TITLE
Использование виджета без локального service.php

### DISCUSSION THREAD

**Discussion by eugenegantz** [cdek-it/widget#54] [General]
Добрый день.

Вопрос.
Существует способ пользоваться виджетом не используя локальный service.php? То есть просто пользоваться вашей реализацией списка ПВЗ на вашей стороне.

P.S.
Я раньше использовал старую версию виджета. Она была довольна простая -- просто подключаешь на страницу скрипт и пользуешься списком ПВЗ. Очень удобно, на мой взгляд, так и должны работать виджеты.

После удаления старой версии, выяснилось, что актуальная версия требует дополнительной реализации на сервере. Компанией CDEK предусмотрен готовый service.php. Но, в случае если магазин реализован на другой платформе, необходимо с нуля реализовывать собственный обработчик на сервере.

Учитывая, что от виджета может требоваться только список ПВЗ, писать собственную реализацию кажется задачей избыточной.

Было бы здорово если бы для установки виджета достаточно было бы просто подключить скрипт на страницу.

**Comment by vermorag** [comment]
Добрый день. О причинах отдельного service.php писал в https://github.com/cdek-it/widget/issues/34#issuecomment-1931214736.
Раньше пользовались тестовой средой, однако в настоящий момент, во избежание утечек данных, она была закрыта.
Использовать виджет полностью без бэкенд части можно будет с версии 4.0.

**Comment by vladkras** [comment]
Хрен редьки не слаще, хранить эти ключи прямо в файле на сервере — тоже преступление против ИБ.
